src/frontend/src/expr_factory.rs

```diff
@@ -76,6 +81,62 @@ impl CreateExprFactory for DefaultCreateExprFactory {
     }
 }
 
+pub(crate) async fn create_external_expr(
+    create: &CreateExternalTable,
+    query_ctx: QueryContextRef,
+) -> Result<CreateTableExpr> {
+    let (catalog_name, schema_name, table_name) =
+        table_idents_to_full_name(&create.name, query_ctx)
+            .map_err(BoxedError::new)
+            .context(error::ExternalSnafu)?;
+    let time_index = "".to_string();
+
+    let mut options = create.options.clone();
+
+    let (object_store, files) = prepare_immutable_file_table(&options)
+        .await
+        .context(error::ParseImmutableTableOptionsSnafu)?;
+
+    let schema = if !create.columns.is_empty() {
+        let columns_schemas: Vec<_> = create
+            .columns
+            .iter()
+            .map(|column| column_def_to_schema(column, false).context(error::ParseSqlSnafu))
+            .collect::<Result<Vec<_>>>()?;
+        RawSchema::new(columns_schemas)
+    } else {
+        let format =
+            parse_immutable_file_table_format(&options).context(error::ParseFileFormatSnafu)?;
+        infer_immutable_file_table_schema(&object_store, format, &files)
+            .await
+            .context(error::InferSchemaSnafu)?
+    };
+
+    let meta = ImmutableFileTableOptions { files };
+
+    // lists files in the frontend to reduce unnecessary scan requests repeated in each datanode.
+    options.insert(
+        IMMUTABLE_TABLE_META_KEY.to_string(),
+        serde_json::to_string(&meta).context(error::EncodeJsonSnafu)?,
+    );
+
+    let expr = CreateTableExpr {
+        catalog_name,
+        schema_name,
+        table_name,
+        desc: "".to_string(),
+        column_defs: column_schemas_to_defs(schema.column_schemas)?,
+        time_index,
```

Comment:
suggestion: Define the variable as close to the place uses it as possible. Actually, we don't need the time_index local variable.
```suggestion
        time_index: "".to_string(),
```